codegen/generate.go

```diff
@@ -91,3 +192,11 @@ func stripCurlyBraces(path string) string {
 	path = strings.ReplaceAll(path, "}", "")
 	return path
 }
+
+// parentDir returns the directory containing the given file.
+// ex. generated/resources/transform-transformation-name.go
+// returns generated/resources/
+func parentDir(pathToFile string) string {
```

Comment:
I like this idea and I've been playing with it a little. I found that with the current way the algorithm works, it will return correct results if `make generate` is run from the home directory or beneath it. It won't work correctly if run from outside the home directory. 

When I played with `filepath.Dir`, I'm not sure there was any benefit. It was still impossible to get the correct `homeDir` from outside the home directory. From inside it, it appeared actually slightly more brittle.

Can you propose the code you're thinking of here? You may have something different in your mind than what I'm fiddling around with.